### PR TITLE
RUBY-1383 RUBY-1854 RUBY-1834 test & fix read preference handling by the driver in various scenarios

### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

**Comment by saghm** [comment]
Does this need to be rebased with master? It looks like it contains all the changes from the `write` -> `write_concern` PR

File: lib/mongo/collection.rb [hunk 1]
```diff
@@ -228,6 +228,13 @@ def capped?
     #
     # @since 2.0.0
     def create(opts = {})
+      # Passing read options to create command causes it to break.
+      # Filter the read options out.
+      # TODO put the list of read options in a class-level constant when
```
Comment (HanaPearlman) [inline lib/mongo/collection.rb]: Do we have this list?

File: lib/mongo/session.rb [hunk 2]
```diff
@@ -770,7 +779,7 @@ def add_txn_num!(command)
     def add_txn_opts!(command, read)
       command.tap do |c|
         # The read preference should be added for all read operations.
-        if read && txn_read_pref = txn_read_preference
+        if false && read && txn_read_pref = txn_read_preference
```
Comment (HanaPearlman) [inline lib/mongo/session.rb]: This statement should never evaluate to true, correct?

Comment (saghm) [inline lib/mongo/session.rb]: Why is this block being disabled? if we don't want it to run, can we just remove it?

File: lib/mongo/collection.rb (hunk 1 above)
Comment (p-mongo) [reply lib/mongo/collection.rb]: No, this will need to be researched or done organically over time.

File: lib/mongo/session.rb (hunk 2 above)
Comment (p-mongo) [reply lib/mongo/session.rb]: Right,  I forgot to remove this piece.

**Pull request merged**